threescale_api/resources.py

```diff
@@ -389,6 +389,13 @@ def __init__(self, *args, entity_name='tenant', entity_collection='tenants', **k
         super().__init__(*args, entity_name=entity_name,
                          entity_collection=entity_collection, **kwargs)
 
+    def read(self, id, **kwargs):
```

Comment:
Would change `id` to `tenant: Union['Tenant', int]` and later extract the id with `_extract_entity_id(tenant)`. It would allow to pass either id directly or also whole Tenant resource.

Reply:
Why would you want to pass Tenant resource? Isn't this method used when trying to retrieve the tenant in the first place?

Reply:
Makes sense. Disregard what I said :D